kademlia/src/test/java/ingsw/group1/kademlia/NodesRoutingTableTest.java

```diff
@@ -0,0 +1,265 @@
+        when(referenceNode.getKey()).thenReturn(new BinarySet(bitSet));
+        distance = getDistancePeerNode(nodeOwner, referenceNode);
+        when(nodeOwner.getDistance(referenceNode)).thenReturn(distance);
+        assertTrue(rt.add(referenceNode));
+
+        bitSet = BitSet.valueOf(new byte[]{(new Integer(2)).byteValue()}); //KEY = 010
+        when(closestNode.getKey()).thenReturn(new BinarySet(bitSet));
+        distance = getDistancePeerNode(nodeOwner, closestNode);
+        when(nodeOwner.getDistance(closestNode)).thenReturn(distance);
+        assertTrue(rt.add(closestNode));
+
+        distance = getDistancePeerNode(referenceNode, closestNode);
+        when(referenceNode.getDistance(closestNode)).thenReturn(distance);
+        distance = getDistancePeerNode(referenceNode, node);
+        distance = getDistancePeerNode(referenceNode, referenceNode);
+
+        assertEquals(closestNode, rt.getClosest(referenceNode));
+    }
+
+    @Test
+    public void getKClosest() {
+        bitSet = BitSet.valueOf(new byte[]{(new Integer(3)).byteValue()}); //KEY = 011
+        when(node.getKey()).thenReturn(new BinarySet(bitSet));
+        distance = getDistancePeerNode(nodeOwner, node);
+        when(nodeOwner.getDistance(node)).thenReturn(distance);
+
+        int numberNodesClosest = 2;
+        ArrayList<Node<BinarySet>> nodesClosest = new ArrayList<>();
+        int key = 1;
+        for (int i = 0; i < numberNodesClosest; i++) {  //1,2,3
+            nodesClosest.add(mock(Node.class));
+            Node<BinarySet> nodeAdded = nodesClosest.get(i);
+            bitSet = BitSet.valueOf(new byte[]{(new Integer(key)).byteValue()});
+            when(nodeAdded.getKey()).thenReturn(new BinarySet(bitSet));
+
+            distance = getDistancePeerNode(nodeOwner, nodeAdded); //from nodeOwner for the add in rt
+            when(nodeOwner.getDistance(nodeAdded)).thenReturn(distance);
+
+            distance = getDistancePeerNode(node, nodeAdded); //from the node to getKClosest
+
+            assertTrue(rt.add(nodeAdded));
+
+            key++;
+        }
+
+        ArrayList<Node<BinarySet>> othersNodes = new ArrayList<>();
+        key += 2;
+        for (int i = 0; i < 2; i++) { //5,6
+            othersNodes.add(mock(Node.class));
+            Node<BinarySet> nodeAdded = othersNodes.get(i);
+            bitSet = BitSet.valueOf(new byte[]{(new Integer(key)).byteValue()});
+            when(nodeAdded.getKey()).thenReturn(new BinarySet(bitSet));
+
+            distance = getDistancePeerNode(nodeOwner, nodeAdded); //from nodeOwner for the add in rt
+            when(nodeOwner.getDistance(nodeAdded)).thenReturn(distance);
+            distance = getDistancePeerNode(node, nodeAdded); //from the node to getKClosest
+
+            key++;
+
+            assertTrue(rt.add(nodeAdded));
+        }
+
+        assertArrayEquals(nodesClosest.toArray(), rt.getKClosest(node));
+    }
+
+    @Test
+    public void getKClosest_NodeOwner() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'getKClosest_NodeOwner' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/RicDeZen/ingsw-group1-kademlia/pullRequest?prid=4745729)